src/pmdas/bpf/bpf.c

```diff
@@ -243,13 +255,14 @@ bpf_load_modules(dict *cfg)
         if (ret != 0) {
             libbpf_strerror(ret, errorstring, 1023);
             pmNotifyErr(LOG_ERR, "module initialisation failed: %s, %d, %s", module_name, ret, errorstring);
-            free(module_name);
+            ret = pmdaCacheStore(CACHE_CLUSTER_IDS, PMDA_CACHE_HIDE, module_name, bpf_module);
```

Comment:
I think you still need to free(module_name) here as pmdaCacheStore will create its own duplicate internally.  Otherwise all looks good!

Reply:
I've free'd `module_name` in line 242, and in line 244 I'm reassigning it to the internal duplicate allocated in `pmdaCacheStore`. I've tried this error case with valgrind (running it as non-root-user triggers this condition), and it doesn't report any leaks.